### ISSUE TITLE
Błąd przy logowania do REST API sandbox

### ISSUE THREAD

**Issue by qshop123** [allegro/allegro-api#8654]
Występuje problem: 
 - [x] z zasobem REST API / with a REST API resource
 - [ ] ze środowiskiem produkcyjnym / with the production environment
 - [ ] ze środowiskiem testowym / with the test environment
 - [ ] z logowaniem lub autoryzacją / with login or authorization
 - [ ] na portalu https://developer.allegro.pl/ / at https://developer.allegro.pl/en/
 - [ ] inny / other

Hejka
Znowu to samo.
1. Mam konto na sandbox Client:44197345
2. Mam połączoną aplikację do konta, Client ID : 07043418877049d09d013c5a65ae7def
3. Loguję się i wyskakuje błąd : Nie możemy wyświetlić strony, Numer błędu: 55829876245f3342

Nowej aplikacji dodać nie mogę, bo mam komunikat 

Konto : 
Nie możesz zarejestrować nowej aplikacji
Aktywuj konto oraz włącz dwustopniowe logowanie. Link prowadzi do produkcji!!! https://allegro.pl/pomoc/dla-kupujacych/logowanie-i-haslo/czym-jest-dwustopniowe-logowanie-i-dlaczego-warto-z-niego-korzystac-dykqg9nMKSZ

Po co w systemie testowym monit o braku aktywacji konta wersji produkcyjnej i link do proda?

**Comment by MaciejFrackowiak** [comment]
Kilka wyjaśnień:
1. Twoje konto Sandbox nie posiada uzupełnionych danych, nie zostało aktywowane od 2019 roku;
2. ClientID, który podajesz jest produkcyjny, a chcesz wywoływać zasoby na Sandbox - tak to nie działa. Musisz zarejestrować osobną aplikację na Sandbox by moć na Sandbox działać;
3. Tak, link prowadzi do produkcji, ponieważ jest to przekierowanie do Pomocy Allegro, gdzie są aktualne informacje wspierające korzystanie z Allegro - nie ma alternatywnej pomocy dla Sandbox;
4. Domyślny kod do aktywacji dwustopniowego logowania na Sandbox to 123456.

**Comment by qshop123** [comment]
Dziękuję za odpowiedź.
1. Możliwe
2. Załącznik
![allegro_info_a](https://github.com/allegro/allegro-api/assets/4890648/0504fec2-aa08-4cd7-b71d-b37bdeb922d5)
3. Dziwne i mylące. Już wiem co zrobiłem, zalogowałem się na sandbox i w zakładce aplikacje https://apps.developer.allegro.pl.allegrosandbox.pl/ kliknąłem po sprawdzeniu aplikacji na logo "Allegro" bo myślałem że mnie przekieruje do https://allegro.pl.allegrosandbox.pl/ tymczasem to logo jest podpięte do "https://allegro.pl/" - i tam znowu wszedłem w aplikacje aby je sprawdzić ale już byłem na produkcji nie na teście...Po co w systemie testowym dajecie linki prowadzące do produkcji? Koduje od 2001 roku i coś takiego mi do głowy nie wpadłoby nigdy.... 
4. Ok

Ale mam inny problem, jak już się zalogowałem to wywołując zasób https://api.allegro.pl.allegrosandbox.pl/sale/shipping-rates?seller.id=44189787 dostaję to

`array (
  'options' => 
  array (
    'http' => 
    array (
      'method' => 'GET',
      'header' => 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX25hbWUiOiI0NDE4OTc4NyIsInNjb3BlIjpbImFsbGVncm86YXBpOm9yZGVyczpyZWFkIiwiYWxsZWdybzphcGk6cHJvZmlsZTp3cml0ZSIsI...
Content-Type: application/vnd.allegro.public.v1+json
Accept: application/vnd.allegro.public.v1+json',
      'content' => '[]',
      'ignore_errors' => true,
    ),
  ),
  'url' => 'https://api.allegro.pl.allegrosandbox.pl/sale/shipping-rates?seller.id=44189787',
  'response' => '<html><body><h1>413 Payload Too Large</h1>
The request entity exceeds the maximum allowed.
</body></html>
',
)`

Tam nie ma jakiejś wielkiej paczki danych. Jest włączone http2 w indiańcu...

Proszę o pomoc.

**Comment by MaciejFrackowiak** [comment]
Przed momentem sprawdziłem działanie zasobu odpowiedzialnego za pobieranie cenników dostawy - w logach widzę same wywołania z 200 (ostatnie 24 godziny). Nie ma ani jednego śladu po Payload Too Large. Co więcej, przy wywoływaniu [GET /sale/shipping-rates](https://developer.allegro.pl/documentation/#operation/getListOfShippingRatestUsingGET) nie musisz już podawać seller.id.

**Comment by qshop123** [comment]
Wywołane przed chwilą "https://api.allegro.pl.allegrosandbox.pl/sale/categories"

'url' => 'https://api.allegro.pl.allegrosandbox.pl/sale/categories', 'response' => '<html><body><h1>413 Payload Too Large</h1>
The request entity exceeds the maximum allowed.</body></html>', 

Taka przychodzi od was odpowiedź....

**Comment by MaciejFrackowiak** [comment]
Proszę podaj trace-id wywołania, ponieważ szukając ręcznie nie znalazłem żadnego wywołania zakończonego błędem 413.

**Comment by qshop123** [comment]
Początek bearera : Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9
Wywołanie dzisiaj na sandboxa o 11:53

**Comment by MaciejFrackowiak** [comment]
Tak nie znajdę Twojego wywołania, stąd trace-id jest niezbędny.

**Comment by qshop123** [comment]
Header

Nie działa : 
[
	'method' => 'GET',
	'header' => 'Authorization: Bearer ....., Content-Type: application/vnd.allegro.public.v1+json, Accept: application/vnd.allegro.public.v1+json',
	'content' => '[]',
	'ignore_errors' => true
]

Działa
[
	'method' => 'GET',
	'header' => 'Authorization: Bearer ....., Content-Type: application/vnd.allegro.public.v1+json, Accept: application/vnd.allegro.public.v1+json',
	'ignore_errors' => true
]

Zrobiliście coś po stronie API że podanie pustej tablicy jako "[]" powoduje że wasze API głupieje i odpowiada "413 Payload Too Large" - jak dla mnie to "413 Payload Too Small" ale mam za mało doświadczenia aby zrobić prawidłową obsługę błędów tak jak wy ...

**Comment by MaciejFrackowiak** [comment]
Zgodnie z dokumentacją dla zasobów GET nie masz przekazywać żadnej wartości.

**Comment by qshop123** [comment]
Tak, lecz to zawsze działało jak się przekazywało puste a od pewnego czasu nie działa...
